crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -1212,35 +1268,41 @@ impl DeltaTableInputEndpointInner {
         path: &str,
         polarity: bool,
         table: &DeltaTable,
+        column_names: &str,
         input_stream: &mut dyn ArrowStream,
         receiver: &mut Receiver<PipelineState>,
-    ) {
+    ) -> AnyResult<()> {
+        let description = format!("file '{path}'");
+
         // See comment about `object_store_url` above.
         let full_path = format!("{}{}", table.log_store().object_store_url().as_str(), path);
-        let df = match self
-            .datafusion
-            .read_parquet(full_path, ParquetReadOptions::default())
-            .await
-        {
-            Err(e) => {
-                self.consumer.error(
-                    true,
-                    anyhow!("error reading Parquet file '{path}' listed in table log: {e}"),
-                );
-                return;
-            }
-            Ok(df) => df,
+
+        // Create a datafusion table backed by these files.
+        let table = Arc::new(
+            self.create_parquet_table(table, vec![full_path.clone()], &description)
+                .await?,
+        );
+
+        self.datafusion.register_table("tmp_table", table).map_err(|e| {
```

Comment:
should I worry about a name collision?

Reply:
Probably not. The datafusion instance is local to each connector instance, and this shouldn't create more than one `tmp_table` at a time.